src/FileStrategy.ts

```diff
@@ -0,0 +1,90 @@
+import GitDB from './gitdb';
+import fse from 'fs-extra';
+import { Filter, SetCallback, DBRecord } from './collection';
+
+interface Update {
+  updated: boolean;
+  filePaths: string[];
+}
+
+interface Delete {
+  deleted: boolean;
+  filePaths: string[];
+}
+
+export class FileStrategy<T extends DBRecord> {
+  private db: GitDB;
+  private name: string;
+
+  constructor(gitDb: GitDB, name: string) {
+    this.db = gitDb;
+    this.name = name;
+  }
+
+  public getAll(): Promise<T[]> {
+    return this.db.readDocuments(this.name);
+  }
+
+  public async getData(callback: Filter<T>): Promise<T[]> {
+    const documents = await this.db.readDocuments(this.name);
```

Comment:
How about creating a `utils/file.js` and have `readDir`, `readFile` alike methods that can be used here? just to avoid dependency on gitDB which is a top level class